File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    const answer = [];
    let count = 0;
    let max = 0;
    
    progresses.forEach((progress, index) => {
        const day = Math.ceil((100 - progress) / speeds[index]);
        
        if (day <= max) count++;
        else {
            max = day;
            count && answer.push(count);
            count = 1;
        }
    });
    
    return [...answer, count];
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
def solution(progresses, speeds):
    answer = []
    _count = 0
    _max = 0
    
    for progress, speed in zip(progresses, speeds):
        _day = -(-(100 - progress) // speed)
        
        if _day <= _max: _count += 1
        else:
            _max = _day
            if _count > 0: answer.append(_count)
            _count = 1
    
    return answer + [_count]